src/components/useBuilds.ts

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright 2020 RoadieHQ
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { errorApiRef, useApi } from '@backstage/core';
+import { useCallback, useState } from 'react';
+import { useAsyncRetry } from 'react-use';
+import { buildKiteApiRef } from '../api';
+import { BuildKiteBuildInfo } from './types';
+
+export const transform = (
+  buildsData: BuildKiteBuildInfo[],
+  restartBuild: { (orgSlug: string, pipelineSlug: string, buildNumber: number): Promise<void> },
+): BuildKiteBuildInfo[] => {
+  return buildsData.map(buildData => {
+    const tableBuildInfo: BuildKiteBuildInfo = {
+      ...buildData,
+      onRestartClick: () => {
+        const splitUrl = buildData.url.split('/');
+        restartBuild(splitUrl[5], buildData.pipeline.slug, buildData.number);
```

Comment:
Could we name these `splitUrl[n]` variables ? 
Alternatively we could write a helper function which parses a build url into named parts.
And I wonder why we don't pass restart the full url (though admittedly this approach is more flexible)